lib/api/core/httpRouter/routeHandler.js

```diff
@@ -34,34 +36,54 @@ const
  * @class RouteHandler
  * @param {string} url - parsed URL
  * @param {object} query - parsed query string
- * @param {object} httpRequest - raw HTTP request informations
- * @throws {BadRequestError} If x-kuzzle-volatile HTTP header can not be parsed in JSON format
+ * @param {HttpMessage} message
+ * @throws {BadRequestError} If x-kuzzle-volatile HTTP header can not be parsed
+ *                           in JSON format
  */
 class RouteHandler {
-  constructor(url, query, httpRequest) {
+  constructor(url, query, message) {
     this.handler = null;
+    this._request = null;
     this.url = url;
-    this.request = null;
-    this.httpRequest = httpRequest;
+
     this.data = {
-      requestId: this.httpRequest.requestId
+      requestId: message.requestId,
+      body:message.content,
+      ...query
     };
 
-    for (const k of Object.keys(this.httpRequest.headers)) {
-      if (k.toLowerCase() === 'authorization'
-        && this.httpRequest.headers[k].startsWith('Bearer ')
+    this.connection = {
+      connection: {
+        id: message.connection.id,
+        protocol: 'http',
+        url: message.url,
+        headers: message.headers,
+        ips: message.ips
+      }
+    };
+
+    for (const k of Object.keys(message.headers)) {
+      if ( k.toLowerCase() === 'authorization'
```

Comment:
Nit
```suggestion
      if (k.toLowerCase() === 'authorization'
```

Reply:
Check our wrapping rules: we add an extra space when conditions are split to align them.